framework/src/base/ComputeJacobianThread.C

```diff
@@ -222,6 +238,46 @@ ComputeJacobianThread::onInternalSide(const Elem *elem, unsigned int side)
 }
 
 void
+ComputeJacobianThread::onInterface(const Elem *elem, unsigned int side, BoundaryID bnd_id)
+{
+  // Pointer to the neighbor we are currently working on.
+  const Elem * neighbor = elem->neighbor(side);
+
+  // Get the global id of the element and the neighbor
+  const dof_id_type
+    elem_id = elem->id(),
+    neighbor_id = neighbor->id();
+
+  if ((neighbor->active() && (neighbor->level() == elem->level()) && (elem_id < neighbor_id)) || (neighbor->level() < elem->level()))
```

Comment:
So... what happens if there are different levels of refinement in the elements on either side of the interface?  Is this all designed to work with adaptivity?